src/components/booking/partial/steps-actions.tsx

```diff
@@ -356,6 +355,58 @@ export const stepsActions = ({
     },
   };
 
+  const saveDataOnValidation = {
+    execute: async (formData: BookingFormData, boat: Boat) => {
```

Comment:
`BookingFormData` 
This is a different form, maybe we can use different type  (Idea, but you may get confuse with typing

Reply:
I just created another type similar to the booking and removed unused attributes